controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1425,6 +1440,92 @@ bool master_thread::handle_intel_slave_join(Socket *sd, ieee1905_1::CmduMessageR
     return true;
 }
 
+bool master_thread::handle_non_intel_slave_join(Socket *sd,
+                                                std::shared_ptr<ieee1905_1::tlvWscM1> tlvwscM1,
+                                                std::string bridge_mac, std::string radio_mac,
+                                                ieee1905_1::CmduMessageTx &cmdu_tx)
+{
+    // TODO backhaul handling.
+    // Multi-AP Agent doesn't say anything about the backhaul, so we have to rely on Intel Slave Join for that.
+    std::string backhaul_mac = "";
+
+    // TODO bridge handling.
+    // Assume repeater
+    beerocks::eType ire_type = beerocks::TYPE_IRE;
+
+    // bridge_mac node may have been created from DHCP/ARP event, if so delete it
+    // this may only occur once
+    if (database.has_node(bridge_mac) && (database.get_node_type(bridge_mac) != ire_type)) {
+        database.remove_node(bridge_mac);
+    }
+    // add new GW/IRE bridge_mac
+    LOG(DEBUG) << "adding node " << bridge_mac << " under " << backhaul_mac << ", and mark as type "
+               << ire_type;
+    database.add_node(bridge_mac, backhaul_mac, ire_type);
+    database.set_node_state(bridge_mac, beerocks::STATE_CONNECTED);
+
+    // Controller expects backhaul which contains bridge which contains radio, but we only get radio...
+
+    // Update existing node, or add a new one
+    if (database.has_node(radio_mac)) {
+        if (database.get_node_type(radio_mac) != beerocks::TYPE_SLAVE) {
+            database.set_node_type(radio_mac, beerocks::TYPE_SLAVE);
+            LOG(ERROR) << "Existing mac node is not TYPE_SLAVE";
+        }
+        database.clear_hostap_stats_info(radio_mac);
+    } else {
+        // TODO Intel Slave Join has separate radio MAC and UID; we use radio_mac for both.
+        database.add_node(radio_mac, bridge_mac, beerocks::TYPE_SLAVE, radio_mac);
+    }
+    database.set_hostap_is_acs_enabled(radio_mac, false);
+    // TODO
```

Comment:
There should be no slave reconf in generic mode - we have WSC autoconfig renew for that, so I guess this TODO should be removed.